systemtests/tests/py3plug-fd-libcloud/testrunner-default

```diff
@@ -64,49 +58,66 @@ rm ${tmp}/data/weird-files/newline*
 rm ${tmp}/data/weird-files/tab*
 # the following file also makes problems
 rm ${tmp}/data/weird-files/filename-with-non-utf8-bytestring*
-# s3cmd does not sync empty dirs
+
+# s3cmd does not sync empty dirs, but you can create them by placing
+# into them an empty dummy file with same name
+# Most end-user expect to be able to restore those empty dirs.
 rmdir ${tmp}/data/weird-files/big-X
 rmdir ${tmp}/data/weird-files/subdir
+mkdir -p ${tmp}/data/empty_subdir/
+touch ${tmp}/data/empty_subdir/empty_subdir
+${S3} sync "${BackupDirectory}" s3://${bucket_name}
+# create directly an empty directory in the s3 (would be a plus) but
+# unfortunately doesn't work
+# ${S3} put remote_empty s3://${bucket_name}/remote_empty/
 
-${S3} sync "$BackupDirectory" s3://$bucket_name
+echo "Data preparation done"
 
 start_test
 
-cat <<END_OF_DATA >$tmp/bconcmds
+cat <<END_OF_DATA >${tmp}/bconcmds
 @$out /dev/null
 messages
-@$out $tmp/log1.out
-setdebug level=100 storage=File
-setdebug level=200 client=bareos-fd trace=1 timestamp=1
-label volume=TestVolume001 storage=File pool=Full
-run job=$JobName yes
+@$out ${tmp}/log1.out
+setdebug level=100 trace=1 timestamp=1 storage=File
+setdebug level=200 trace=1 timestamp=1 client=bareos-fd
+run job=${JobName} yes
+wait
+setdebug level=0 client=bareos-fd
 status director
 status client
 status storage=File
 wait
 messages
-@#
-@# now do a restore
-@#
-@$out $tmp/log2.out
-wait
-restore client=bareos-fd fileset=PluginTest where=$tmp/bareos-restores select all done
-yes
+END_OF_DATA
+
+echo "Full backup stage"
+run_bconsole
+expect_grep "Backup OK" "${tmp}/log1.out" "Full Backup not found!"
+if [ ${estat} -ne 0 ]; then
+    exit ${estat}
+fi
+
+cat <<END_OF_DATA >${tmp}/bconcmds
+@$out ${tmp}/log2.out
+setdebug level=200 trace=1 timestamp=1 client=bareos-fd
+restore client=bareos-fd fileset=PluginTest where=${tmp}/bareos-restores select all done yes
 wait
 messages
 quit
 END_OF_DATA
+echo "Restore backup stage"
+run_bconsole
+expect_grep "Restore OK" "${tmp}/log2.out" "Full Restore not ok!"
+if [ ${estat} -ne 0 ]; then
+    exit ${estat}
+fi
```

Comment:
exiting here means that you will skip stopping the minio server.  Maybe its better to move to the following structure:
- test-setup: setup data, start minio, create bucket
- testrunner-backup: create backup
- testrunner-restore: restore backup & check everything
- test-cleanup: delete bucket, stop minio, delete all data (like normal test-cleanup)

Reply:
Letting minio running is the expected state at this point. This allow interactive debug if the test is failing.
But I agree I will extent the test-cleanup for that.